### PR TITLE
admin-module-list-users-delete-user

### PR DESCRIPTION
**Pull request by dungvan**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\User;
+use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
```
Comment (at-duyho) [inline app/Http/Controllers/UserController.php]: -you should create constructor to use DI in laravel.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -19,4 +21,27 @@ public function index()
...
+     *
+     * @param Integer $id id of user to destroy
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+
+        if (Auth::user()->id == $id) {
+            flash('can not delete yourself!')->error();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: define message to lang file.

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -19,4 +21,27 @@ public function index()
...
+        } else {
+            if (User::findOrFail($id)->delete()) {
+                flash('delete successfully!')->success();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -define message to lang file

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -19,4 +21,27 @@ public function index()
...
+            } else {
+                flash('can not delete yourself!')->error();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: -define message to lang file

**Review by ghost — CHANGES_REQUESTED** [review]
check comment.please!

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -14,9 +28,32 @@ class UserController extends Controller
     public function index()
     {
         if (!isset($listUsers)) {
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: what's $listUsers?

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -14,9 +28,32 @@ class UserController extends Controller
...
+     *
+     * @param Integer $id id of user to destroy
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+
+        if (Auth::user()->id == $id) {
+            flash(trans('user/message.delete-error'))->error();
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: you should use user::message.delete-error instead of user/message.delete-error.

**Review by ghost — CHANGES_REQUESTED** [review]
pleas check comment.

File: app/Http/Controllers/UserController.php (hunk 6 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: i just do this, and it not work!!

File: app/Http/Controllers/UserController.php (hunk 5 above)
Comment (dungvan) [reply app/Http/Controllers/UserController.php]: this is a list of users get from database and return to index to show the users record

File: app/Http/Controllers/UserController.php [hunk 7]
```diff
@@ -3,20 +3,55 @@
...
      * Display a listing of the resource.
      *
      * @return \Illuminate\Http\Response
      */
     public function index()
     {
-        if (!isset($listUsers)) {
-            $listUsers = User::all();
+        $users = $this->user->paginate(10);
+
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: remove blank line.

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -3,20 +3,55 @@
...
+    /**
+     * Destroy user
+     *
+     * @param Integer $id id of user to destroy
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: remove blank line.

File: app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -3,20 +3,55 @@
...
+        if (Auth::user()->id == $id) {
+            flash(trans('user.delete-error'))->error();
+        } else {
+            if ($this->user->findOrFail($id)->delete()) {
+                flash(trans('user.delete-success'))->success();
+            } else {
+                flash(trans('user.delete-error'))->error();
+            }
         }
```
Comment (ghost) [inline app/Http/Controllers/UserController.php]: remove blank line.

File: composer.json [hunk 10]
```diff
@@ -7,6 +7,7 @@
     "require": {
         "php": ">=5.6.4",
         "intervention/image": "^2.4",
+        "laracasts/flash": "^3.0",
```
Comment (ghost) [inline composer.json]: why this pull request like https://github.com/AT-PHPIntership/food-market-order/pull/72/files

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment.

File: composer.json (hunk 10 above)
Comment (dungvan) [reply composer.json]: this is list and delete module, the pull  https://github.com/AT-PHPIntership/food-market-order/pull/72/files
is create module, it different!

**Comment by dungvan** [comment]
check my comment! thank you!

File: resources/views/users/index.blade.php [hunk 11]
```diff
@@ -0,0 +1,70 @@
+@extends('layouts.master')
+
+@section('main-content')
+
+    @include('flash::message')
+
+    @if(!isset($users))
```
Comment (ghost) [inline resources/views/users/index.blade.php]: why do you use negative condition?  for example:
if (positive condition) {
} else {
}

**Review by ghost — CHANGES_REQUESTED** [review]
Check comment.

File: resources/views/users/index.blade.php (hunk 11 above)
Comment (dungvan) [reply resources/views/users/index.blade.php]: if the $users is null, the view just view `Nothing to show!` in if statement, it easy to read the code, i think the positive have the same thing, but the line `Nothing to show!` in the bottom, it so hard to view the code!

**Comment by dungvan** [comment]
check my comment! please!

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**